crates/bevy_app/src/app.rs

```diff
@@ -867,6 +867,28 @@ impl App {
         self
     }
 
+    /// Adds the type data `D` to type `T` in the [TypeRegistry](bevy_reflect::TypeRegistry) resource
```

Comment:
I'd like to see some brief compare-and-contrast between this and its sibling to help guide users.